fix(microsoft-defender-intel): connector crashes on every event with TypeError, HttpUrl + str, during OAuth token acquisition (regression in 7.260520.0)

## Description
The microsoft-defender-intel connector fails on every stream event when attempting to push indicators to Microsoft Defender. The failure happens during OAuth token acquisition, before any Defender API call is made, so no indicators are pushed at all. Credentials are valid and are not the cause.

Every event fails with:

`TypeError: unsupported operand type(s) for +: 'HttpUrl' and 'str'`

The exception is raised in api_handler.py line 78, in _get_authorization_header, at:

`"scope": self.base_url + "/.default"`

<img width="768" height="526" alt="Image" src="https://github.com/user-attachments/assets/a374918f-9615-447f-b3b7-445cb23b9914" />

Diagnosis: settings.py types base_url as a pydantic HttpUrl. Since [PR #6394](https://github.com/OpenCTI-Platform/connectors/pull/6394) / commit 58688b5e ("[all] Fix url to avoid double slash", merged 2026-05-20), the str() coercion on base_url was removed, and the isinstance(base_url, str) check is always False because the value is always an HttpUrl object. The HttpUrl object then reaches line 78, where it cannot be concatenated with a str.

There is no configuration value that avoids the failure. base_url is force-parsed to HttpUrl regardless of input. In practice this surfaces after a redeploy (for example after a credential change) pulls a newer image.

## Environment

Version: connector image tag latest, affected versions 7.260520.0 onward including master as of 2026-07-13; OpenCTI platform 7.26x

## Reproducible steps

Steps to create the smallest reproducible scenario:

1. Deploy the microsoft-defender-intel connector at version 7.260520.0 or later (or image tag latest) with valid Defender credentials and any valid base_url value.
2. Trigger any stream event that causes the connector to create an indicator in Defender (for example, create an Indicator in OpenCTI within the connector's live stream scope).
3. Observe the connector logs. Every event fails with the TypeError below, and no request ever reaches the Defender API.

## Expected output
The connector acquires an OAuth token and pushes the indicator to Microsoft Defender.

## Actual output
Every event fails at token acquisition:
`[ERROR] Failed processing data {unsupported operand type(s) for +: 'HttpUrl' and 'str'}`


## Additional information

<!-- Any additional information, including logs or screenshots if you have any. -->

--- a/stream/microsoft-defender-intel/src/microsoft_defender_intel_connector/api_handler.py
+++ b/stream/microsoft-defender-intel/src/microsoft_defender_intel_connector/api_handler.py
@@ -52,7 +52,7 @@ class DefenderApiHandler:
         self.tenant_id = tenant_id
         self.client_id = client_id
         self.client_secret = client_secret
-        self.base_url = base_url.rstrip("/") if isinstance(base_url, str) else base_url
+        self.base_url = str(base_url).rstrip("/")
         self.resource_path = resource_path
         self.action = action
         self.expired_after = expired_after
@@ -188,7 +188,9 @@ class DefenderApiHandler:
         Get Threat Intelligence Indicators from Defender.
         :return: List of Threat Intelligence Indicators if request is successful, None otherwise
         """
-        data = self._send_request("get", f"{self.base_url}{self.resource_path}")
+        data = self._send_request(
+            "get", f"{self.base_url}/{self.resource_path.lstrip('/')}"
+        )
         result = data["value"]
         while "@odata.nextLink" in data and data["@odata.nextLink"] is not None:
             data = self._send_request("get", data["@odata.nextLink"])
@@ -203,7 +205,7 @@ class DefenderApiHandler:
         """
         params = f"$filter=indicatorValue eq '{value}'"
         data = self._send_request(
-            "get", f"{self.base_url}{self.resource_path}", params=params
+            "get", f"{self.base_url}/{self.resource_path.lstrip('/')}", params=params
         )
         result = data["value"]
         while "@odata.nextLink" in data and data["@odata.nextLink"] is not None:
@@ -221,7 +223,7 @@ class DefenderApiHandler:
         request_body_observable = self._build_request_body(observable, defender_id)
         data = self._send_request(
             "post",
-            f"{self.base_url}{self.resource_path}",
+            f"{self.base_url}/{self.resource_path.lstrip('/')}",
             json=request_body_observable,
         )
         return data
@@ -240,7 +242,7 @@ class DefenderApiHandler:
 
         data = self._send_request(
             "post",
-            f"{self.base_url}{self.resource_path}/import",
+            f"{self.base_url}/{self.resource_path.strip('/')}/import",
             json=request_body,
         )
         return data
@@ -254,7 +256,7 @@ class DefenderApiHandler:
         request_body = {"IndicatorIds": indicators_ids}
         self._send_request(
             "post",
-            f"{self.base_url}{self.resource_path}/BatchDelete",
+            f"{self.base_url}/{self.resource_path.strip('/')}/BatchDelete",
             json=request_body,
         )
         return True
@@ -267,6 +269,6 @@ class DefenderApiHandler:
         """
         self._send_request(
             "delete",
-            f"{self.base_url}{self.resource_path}/{indicator_id}",
+            f"{self.base_url}/{self.resource_path.strip('/')}/{indicator_id}",
         )
         return True

